### DISCUSSION TITLE
How can i run a SmartNode?

### DISCUSSION THREAD

**Discussion by Demuzx** [thetaspere/theta#27] [Q&A] [answered]
How can i run a SmartNode? What do i need to run a SmartNode, except 600K FITA?

**Comment by thetaspere** [comment]
Please use this documentation for now, our documentation is still on progress
https://docs.raptoreum.com/docs/smartnodes/intro/

**Reply by Demuzx** [reply]
If i use this documentation i'm afraid to lose my 600K FITA. That's why i think i'd better wait for your documentation.

**Reply by thetaspere** [reply]
You will not lose it, in creating a masternode, you will only be asked to transfer to your own address, not send to another wallet address.

**Reply by thetaspere** [reply]
i will write it here, please wait

**Comment by thetaspere** [comment, marked as answer]
1. Enable UFW & Open Ports

```
apt install ufw -y
ufw default deny incoming
ufw default allow outgoing
ufw allow ssh
ufw allow 9933/tcp
ufw allow 9932/tcp
ufw enable
```

2. Sync local wallet and send collateral

3. Start wallet and let it finish syncing.

4. Encrypt wallet (Settings > Encrypt Wallet)

5. Create new receiving address Backup wallet.dat (File > Backup Wallet).
Store in multiple locations. Dump private key for receiving address (this should be printed and stored offline in multiple locations):

6. Send 600k FITA to self
Wait for 2 confirmations
`walletpassphrase yourpasswordhere 600`    from the debug console to unlock the wallet before running the protx setup script.

7. Build protx command for control wallet
An example protx quick_setup command
`protx quick_setup d7898f44c6bdc39a49b0917d5c6bcb61455270ddb9e7eb1168a6b745434daf26 1 194.113.73.87:9933 16xwCzA8pZD5poCWnAaRZjjxdVg5HwZctu`

The structure from left to right is:

Transaction ID: In your wallet go to “Transactions” right click the one you sent yourself earlier and “Copy Transaction ID”. Replace the Transaction ID in example.

Collateral index:
Tools > Debug console. Type smartnode outputs to check if 1 or 0. Adjust example command if needed.

Your smartnode server IP and port. Replace example IP with your Smartnode server IP, leave port as is.

Fee address: This is any address in your wallet which contains enough FITA to pay the fee (cannot be the address which you sent the 600k FITA to). When you enter the protx quick_setup command it is a transaction and needs to be paid for. It is a very small amount 1/2 a FITA is more than enough. In Debug console do “listaddressbalances” to show all addresses with a balance, choose one and replace address in example command.!

```
note
   Enter the protx quick_setup command in Debug console. This will create a .conf file for that node in the same directory you ran the wallet from. Open it and copy the contents
```

Back to the smartnode server
`~./theta-cli stop`
`nano ~/.thetacore/theta.conf`

Paste in what you copied from the .conf file made during the protx command, save and exit.

Start thetad:

`~./thetad`

Wait two minutes and do:

`~./theta-cli smartnode status`
 
This should return:

Ready Ready

 
 
Extras:
Check the node’s current height with:
./theta-cli getblockcount

./theta-cli -getinfo

To update a node that has been PoSe Banded:
protx update_service “protxhash” “ip”:9933 “BLSKEY”

You can get the proTxHash by right clicking your Smartnode in the QT wallet Smartnode list
BLS private key. This is stored on your Smartnode in ~/.thetacore/theta.conf
ie: protx update_service 7d62ce3bcf71825te84y5ca4b93a53583a8f1c9c70989e351c0f470088729a83 yoursmartnodeip:9933 3808471d6e52fe3b683474075cdf50192babe3b38203187ec0cg18fg54e439a4

**Reply by Demuzx** [reply]
Thank you, i'll try it.

**Reply by Silasminer** [reply]
Hello 
I have Following error: 
bad-protx-ipaddr (code 16) (code -1)
an ideas ? i think everything went fine ... but maybe I missed something

Thanks for the help

Silas
